pkg/remote/aws/vpc_security_group_rule_details_fetcher.go

```diff
@@ -20,43 +21,43 @@ func NewVPCSecurityGroupRuleDetailsFetcher(provider terraform.ResourceReader, de
 }
 
 func (r *VPCSecurityGroupRuleDetailsFetcher) ReadDetails(res resource.Resource) (resource.Resource, error) {
-	attrs := make(map[string]interface{})
+	attrs := make(map[string]string)
 
 	if v, ok := res.Attributes().Get("type"); ok {
-		attrs["type"] = v
+		attrs["type"] = fmt.Sprintf("%s", v)
 	}
 	if v, ok := res.Attributes().Get("protocol"); ok {
-		attrs["protocol"] = v
+		attrs["protocol"] = fmt.Sprintf("%s", v)
 	}
 	if v, ok := res.Attributes().Get("from_port"); ok {
-		attrs["from_port"] = int(v.(float64))
+		attrs["from_port"] = fmt.Sprintf("%d", int(v.(float64)))
 	}
 	if v, ok := res.Attributes().Get("to_port"); ok {
-		attrs["to_port"] = int(v.(float64))
+		attrs["to_port"] = fmt.Sprintf("%d", int(v.(float64)))
 	}
 	if v, ok := res.Attributes().Get("security_group_id"); ok {
-		attrs["security_group_id"] = v
+		attrs["security_group_id"] = fmt.Sprintf("%s", v)
 	}
 	if v, ok := res.Attributes().Get("self"); ok {
-		attrs["self"] = v
+		attrs["self"] = fmt.Sprintf("%v", v.(bool))
```

Comment:
Should be `%t` no ?

Reply:
https://github.com/22Acacia/terraform/blob/v0.6.4/flatmap/flatten.go#L31

Reply:
Yeah I think it's better

from the docs;

`%t	the word true or false`